triggering_test.go

```diff
@@ -456,3 +583,21 @@ func BenchmarkLevelTrigger0TriggersOpsAreSamples(b *testing.B) {
 	}
 
 }
+
+func TestKinkModel(t *testing.T) {
+	xdata := []float64{0, 1, 2, 3, 4, 5, 6, 7}
+	ydata := []float64{0, 0, 0, 0, 1, 2, 3, 4}
+	ymodel, a, b, c, X2, err := kinkModelResult(3, xdata, ydata)
+	if a != 0 || b != 0 || c != 1 || X2 != 0 || err != nil {
+		t.Errorf("a %v, b %v, c %v, X2 %v, err %v, ymodel %v", a, b, c, X2, err, ymodel)
+	}
+	ymodel, a, b, c, X2, err = kinkModelResult(4, xdata, ydata)
+	if a != 0.6818181818181821 || b != 0.22727272727272738 ||
+		c != 1.1363636363636362 || X2 != 0.45454545454545453 || err != nil {
```

Comment:
It seems sketchy to do tests on floating point equality. I don't see any go-supported test for approximate equality, though, not in the standard libraries. Not sure what to suggest.

Reply:
I agree in principle, but this work, and is simple, and they don't provide anything to help. I guess we could write our own approx tester or find one in a package, but lets wait till its a problem.